arb_os/precompiles.mini

```diff
@@ -256,6 +272,45 @@ public impure func precompile_0x05() {  // modexp
     evmOp_revert_knownPc(2, 0, 0);
 }
 
+func precompile_0x05_pluggable(
+    args: (EvmCalFrame, ByteArray, any),
+    state: any,
+    firstTime: bool
+) -> (any, any) {  // return ((success: bool, returnData: ByteArray), any)
+    let (topFrame, calldata, _) = args;
+
+    let baseLen = bytearray_get256(calldata, 0);
+    let expLen = bytearray_get256(calldata, 32);
+    let modLen = bytearray_get256(calldata, 64);
+
+    let base = bytearray_extract(calldata, 96, baseLen);
+    let exp = bytearray_extract(calldata, 96+baseLen, expLen);
+    let mod = bytearray_extract(calldata, 96+baseLen+expLen, modLen);
+
+    if let Some(result) = biguint_modexp(
+        biguint_fromBytes(base),
+        biguint_fromBytes(exp),
+        biguint_fromBytes(mod),
+    ) {
```

Comment:
Is the thought that all of the function calls would be compiled as part of the code segment for this pluggable, or would it use the existing implementation?

Reply:
At present a newly uploaded pluggable would have to have its own bytearray library compiled in.  In the future it would be nice to have linkage information available dynamically to pluggables via a lookup table.  But that's too heavy a lift to do at the moment.